test262_errors.txt

```diff
@@ -322,54 +322,228 @@ test262/test/built-ins/Iterator/prototype/take/underlying-iterator-closed-in-par
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Thai.js:24: strict mode: Test262Error: `\P{Script_Extensions=Thai}` should match U+0002BC (`ʼ`)
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Tibetan.js:29: Test262Error: `\P{Script_Extensions=Tibetan}` should match U+003008 (`〈`)
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Tibetan.js:29: strict mode: Test262Error: `\P{Script_Extensions=Tibetan}` should match U+003008 (`〈`)
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Tifinagh.js:26: Test262Error: `\P{Script_Extensions=Tifinagh}` should match U+000302 (`̂`)
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Tifinagh.js:26: strict mode: Test262Error: `\P{Script_Extensions=Tifinagh}` should match U+000302 (`̂`)
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Toto.js:23: Test262Error: `\P{Script_Extensions=Toto}` should match U+0002BC (`ʼ`)
+test262/test/built-ins/RegExp/property-escapes/generated/Script_Extensions_-_Toto.js:23: strict mode: Test262Error: `\P{Script_Extensions=Toto}` should match U+0002BC (`ʼ`)
+test262/test/built-ins/RegExp/property-escapes/generated/Sentence_Terminal.js:104: Test262Error: `\P{Sentence_Terminal}` should match U+001B4E (`᭎`)
+test262/test/built-ins/RegExp/property-escapes/generated/Sentence_Terminal.js:104: strict mode: Test262Error: `\P{Sentence_Terminal}` should match U+001B4E (`᭎`)
+test262/test/built-ins/RegExp/property-escapes/generated/Terminal_Punctuation.js:131: Test262Error: `\p{Terminal_Punctuation}` should match U+000836 (`࠶`)
+test262/test/built-ins/RegExp/property-escapes/generated/Terminal_Punctuation.js:131: strict mode: Test262Error: `\p{Terminal_Punctuation}` should match U+000836 (`࠶`)
+test262/test/built-ins/RegExp/property-escapes/generated/Uppercase.js:16: Test262Error: `\P{Uppercase}` should match U+001C89 (`Ᲊ`)
+test262/test/built-ins/RegExp/property-escapes/generated/Uppercase.js:16: strict mode: Test262Error: `\P{Uppercase}` should match U+001C89 (`Ᲊ`)
+test262/test/built-ins/RegExp/property-escapes/generated/XID_Continue.js:16: Test262Error: `\P{XID_Continue}` should match U+000897 (`ࢗ`)
+test262/test/built-ins/RegExp/property-escapes/generated/XID_Continue.js:16: strict mode: Test262Error: `\P{XID_Continue}` should match U+000897 (`ࢗ`)
+test262/test/built-ins/RegExp/property-escapes/generated/XID_Start.js:16: Test262Error: `\P{XID_Start}` should match U+001C89 (`Ᲊ`)
+test262/test/built-ins/RegExp/property-escapes/generated/XID_Start.js:16: strict mode: Test262Error: `\P{XID_Start}` should match U+001C89 (`Ᲊ`)
+test262/test/built-ins/RegExp/unicode_full_case_folding.js:20: Test262Error: \u1fd3 does not match \u0390
+test262/test/built-ins/RegExp/unicode_full_case_folding.js:20: strict mode: Test262Error: \u1fd3 does not match \u0390
```

Comment:
The upgrade to 15.1 fixed more tests than it broke (16.0 is a different story) but special-casing codepoints U+FB05, U+1FD3 and U+1FE3 has the downside that regex properties like `\p{Changes_When_Casefolded}` don't work.

I'm not sure how to best fix that. I spent quite some time on it but... they're just different enough that they don't play nice with the existing algorithms and data structures.

Reply:
I managed to fix this by retrofitting those special-cased code points into the existing RUN_TYPE encodings.

The thing that threw me off is that it works differently between 15.1 and 16.0, making me think it didn't work at all when in fact it only didn't work with 15.1 (and even then it still works _most_ of the time, see test262_errors.txt for that commit.)